artiq/frontend/artiq_flash.py

```diff
@@ -52,20 +51,24 @@ def get_argparser():
                         type=str, default=None,
                         help="SSH host to jump through")
     parser.add_argument("-t", "--target", default="kasli",
+                        choices=["kasli", "sayma", "kc705"],
                         help="target board, default: %(default)s, one of: "
-                             "kasli sayma kc705")
+                             "%(choices)s")
     parser.add_argument("-V", "--variant", default=None,
-                        help="board variant")
+                        help="board variant (target-dependent)")
```

Comment:
Maybe ``(default: target-dependent)`` ?

Reply:
It would be nice to factor out the ``config`` var so the argparser could access it and auto-populate these defaults/choices, because its proper operation is really not clear if you're running this from the CLI. It effectively requires users to look into the code to understand usage.

Barring that large-ish refactor, I will use ``(default is target-dependent)``.

Reply:
How do you want this to work?